spec/unit/pops/types/p_binary_type_spec.rb

```diff
@@ -0,0 +1,218 @@
+        $x = Binary({value => [251, 239, 255]})
+        notice(assert_type(Binary, $x))
+      CODE
+      expect(eval_and_collect_notices(code)).to eql(['++//'])
+    end
+
+    it "can be created from an Base64 using URL safe encoding by specifying '%u' format'" do
+      # the text 'binar' needs padding with '='
+      code = <<-CODE
+        $x = Binary('--__', '%u')
+        notice(assert_type(Binary, $x))
+      CODE
+      expect(eval_and_collect_notices(code)).to eql(['++//'])
+    end
+
+    it "when created with URL safe encoding chars in '%b' format, these are skipped" do
+      code = <<-CODE
+        $x = Binary('--__YmluYXJ5', '%b')
+        notice(assert_type(Binary, $x))
+      CODE
+      expect(eval_and_collect_notices(code)).to eql(['YmluYXJ5'])
+    end
+
+    it "will error in strict format if string contains URL safe encoded chars" do
+      code = <<-CODE
+        $x = Binary('--__YmluYXJ5', '%B')
+        notice(assert_type(Binary, $x))
+      CODE
+      expect { eval_and_collect_notices(code) }.to raise_error(/invalid base64/)
+    end
+
+    [   '<',
+        '<=',
+        '>',
+        '>='
+    ].each do |op|
+      it "cannot be compared to another instance for magnitude using #{op}" do
+        code = <<-"CODE"
+          $x = Binary('YmluYXJ5')
+          $y = Binary('YmluYXJ5')
+          $x #{op} $y
+        CODE
+        expect { eval_and_collect_notices(code)}.to raise_error(/Comparison of: Binary #{op} Binary, is not possible/)
+      end
+    end
+
+    it 'can be matched against a Binary in case expression' do
+      code = <<-CODE
+        case Binary('YmluYXJ5') {
+          Binary('YWxpZW4='): {
+            notice('nope')
+          }
+          Binary('YmluYXJ5'): {
+            notice('yay')
+          }
+          default: {
+            notice('nope')
+          }
+        }
+      CODE
+      expect(eval_and_collect_notices(code)).to eql(['yay'])
+    end
+
+    it "can be matched against a Binary subsequence using 'in' expression" do
+      # finding 'one' in 'one two three'
+      code = <<-CODE
+        notice(Binary("b25l") in Binary("b25lIHR3byB0aHJlZQ=="))
+        notice(Binary("c25l") in Binary("b25lIHR3byB0aHJlZQ=="))
+      CODE
+      expect(eval_and_collect_notices(code)).to eql(['true', 'false'])
+    end
+
+    it "can be matched against a byte value using 'in' expression" do
+      # finding 'e' (ascii 101) in 'one two three'
+      code = <<-CODE
+        notice(101 in Binary("b25lIHR3byB0aHJlZQ=="))
+        notice(101.0 in Binary("b25lIHR3byB0aHJlZQ=="))
+        notice(102 in Binary("b25lIHR3byB0aHJlZQ=="))
+      CODE
+      expect(eval_and_collect_notices(code)).to eql(['true', 'true', 'false'])
+    end
+
+    it "has a length method in ruby returning the length measured in bytes" do
+      # \u{1f452} is "woman's hat emoji - 4 bytes in UTF-8"
+      a_binary = PBinaryType::Binary.new("\u{1f452}")
+      expect(a_binary.length).to be(4)
+    end
+  end
+
+end
+end
+end
```

Comment:
It's been added. See [PUP-6706](https://tickets.puppetlabs.com/browse/PUP-6706).